extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertyBuildItemCandidateUtil.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.arc.deployment.configproperties;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+public class ConfigPropertyBuildItemCandidateUtil {
+
+    /**
+     * This method inspects the {@code configClass} bytecode to identify all fields that have a default value set in the class
+     * constructor. These fields are removed from the {@link ConfigPropertyBuildItemCandidate} list because we don't want to
+     * throw an exception if no config property value was provided for them. There is no bytecode modification performed during
+     * this process.
+     */
+    public static void removePropertiesWithDefaultValue(String configClass, List<ConfigPropertyBuildItemCandidate> candidates) {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
```

Comment:
We now have a better option that the TCCL (which has always been a little fuzzy): Check out `DeploymentClassLoaderBuildItem` and specifically how it's being used - I think that the way it's being usef for `smallrye-reactive-messaging` should apply here

Reply:
Classloader replaced. I didn't know about `DeploymentClassLoaderBuildItem`, thanks for this very useful comment!

Reply:
Yeah, it's pretty new and definitely very useful!